orc8r/cloud/go/syncstore/writer.go

```diff
@@ -0,0 +1,240 @@
+	rootDigest := digests.GetRootDigest().GetMd5Base64Digest()
+	leafDigestsToSerialize := &protos.LeafDigestsToSerialize{Digests: digests.GetLeafDigests()}
+	leafDigests, err := proto.Marshal(leafDigestsToSerialize)
+	if err != nil {
+		return errors.Wrapf(err, "marshal leaf digests for network %+v", network)
+	}
+	now := clock.Now().Unix()
+
+	txFn := func(tx *sql.Tx) (interface{}, error) {
+		_, err := l.builder.
+			Insert(digestTableName).
+			Columns(nidCol, rootDigestCol, leafDigestsCol, lastUpdatedTimeCol).
+			Values(network, rootDigest, leafDigests, now).
+			OnConflict(
+				[]sqorc.UpsertValue{
+					{Column: rootDigestCol, Value: rootDigest},
+					{Column: leafDigestsCol, Value: leafDigests},
+					{Column: lastUpdatedTimeCol, Value: now},
+				},
+				nidCol,
+			).
+			RunWith(tx).
+			Exec()
+		if err != nil {
+			return nil, errors.Wrapf(err, "insert digests for network %+v", network)
+		}
+		return nil, nil
+	}
+	_, err = sqorc.ExecInTx(l.db, nil, nil, txFn)
+	return err
+}
+
+func (l *syncStore) UpdateCache(network string) (CacheWriter, error) {
+	// the temporary table is namespaced with the unique id of the cacheWriter
+	writerId := fmt.Sprintf("cache_writer_%s", strings.Replace((&storage.UUIDGenerator{}).New(), "-", "_", -1))
+	txFn := func(tx *sql.Tx) (interface{}, error) {
+		_, err := l.builder.CreateTable(writerId).
+			IfNotExists().
+			Column(nidCol).Type(sqorc.ColumnTypeText).NotNull().EndColumn().
+			Column(idCol).Type(sqorc.ColumnTypeText).NotNull().EndColumn().
+			Column(objCol).Type(sqorc.ColumnTypeBytes).NotNull().EndColumn().
+			PrimaryKey(nidCol, idCol).
+			RunWith(tx).
+			Exec()
+		return nil, errors.Wrap(err, "create cached objs tmp table")
+	}
+	_, err := sqorc.ExecInTx(l.db, nil, nil, txFn)
+	if err != nil {
+		return nil, err
+	}
+
+	// The creation time of cacheWriters is tracked by the store for garbage collection
+	store, err := l.fact.StartTransaction(&storage.TxOptions{ReadOnly: true})
+	if err != nil {
+		return nil, errors.Wrapf(err, "error starting transaction")
+	}
+	defer store.Rollback()
+
+	creationTime := make([]byte, 8)
+	binary.LittleEndian.PutUint64(creationTime, uint64(clock.Now().Unix()))
+	err = store.CreateOrUpdate(network, blobstore.Blobs{{
+		Type:  cacheWriterBlobstoreType,
+		Key:   writerId,
+		Value: creationTime,
+	}})
+	if err != nil {
+		return nil, errors.Wrapf(err, "set start time of network %+v, cachewriter %+v in blobstore", network, writerId)
+	}
+	err = store.Commit()
+	if err != nil {
+		return nil, err
+	}
+
+	l.cacheWriters[writerId] = NewCacheWriter(network, writerId, l.db, l.builder)
+	return l.cacheWriters[writerId], nil
+}
+
+func (l *syncStore) RecordResync(network string, gateway string, t uint64) error {
+	store, err := l.fact.StartTransaction(&storage.TxOptions{ReadOnly: true})
+	if err != nil {
+		return errors.Wrapf(err, "error starting transaction")
+	}
+	defer store.Rollback()
+
+	lastResyncBytes := make([]byte, 8)
```

Comment:
Looks like `PutUint64` requires a slice with a fixed length to put the value in, so this might not work @hcgatewood

Reply:
Yeah so make your own `PutUint64` method (either private and local to this file, or public and in the `math` pkg we created recently), which makes it a one-liner -- it will basically be a ~3 line wrapper around the stdlib putuint